net-wireless/bluetuith/bluetuith-0.1.7.ebuild

```diff
@@ -0,0 +1,57 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit go-module
+
+DESCRIPTION="A TUI bluetooth manager for Linux written in Go"
+HOMEPAGE="https://darkhz.github.io/bluetuith"
+
+GIT_DOCUMENTATION_COMMIT="9c215bb1a64bdbf0f88060db4de6701215799033"
+
+if [[ ${PV} == *9999* ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/darkhz/bluetuith.git"
+else
+	SRC_URI="https://github.com/darkhz/bluetuith/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+	SRC_URI+=" https://github.com/rahilarious/gentoo-distfiles/releases/download/${P}/deps.tar.xz -> ${P}-deps.tar.xz"
+	SRC_URI+=" https://github.com/darkhz/bluetuith/archive/${GIT_DOCUMENTATION_COMMIT}.tar.gz -> ${P}-docs.tar.gz"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="Apache-2.0 BSD-2 BSD MIT"
+SLOT="0"
+
+IUSE="doc"
+RESTRICT="test"
+RDEPEND="
+	net-wireless/bluez
+"
+
+if [[ ${PV} == *9999* ]]; then
+	src_unpack() {
+		git-r3_src_unpack
+		git clone --depth 1 -b documentation "${EGIT_REPO_URI}" "${PN}-${GIT_DOCUMENTATION_COMMIT}" || die
```

Comment:
Try git-r3_checkout instead?

Reply:
Also, please hoist the `if` inside the phase. I don't like conditional phase definitions and they may confuse tools like pkgcheck.

Reply:
If I  do it inside phase then that'll define phase for non-live ebuild & that'll replace go-module's src_unpack. 

If you insist for pkgcheck then solution might be to manually trigger go-module_src_unpack

Reply:
I'm afraid I do need to insist -- the reason being, we need some additional rules on top of what is legal in Bash for ebuilds to both be readable but also to allow static analysis in things like pkgcheck without worrying about every possible construct.